### PR TITLE
[WIP] Showcasing crate re-export rename

### PR DESCRIPTION
**Pull request by jojolepro**
## Description

This would slightly reduce the cognitive load of the users, because they know what they want, but maybe not what its called exactly.

On the other hand, (I think) users will be less likely to remember the name of those reexported crates, in case they want to look up the documentation. Though we do re-export those documentations.

Is this wanted?

(Not to be merged now. Doesn't build)

## Modifications

- Renamed amethyst::core::specs to amethyst::core::ecs
- Renamed amethyst::core::nalgebra to amethyst::core::math

## PR Checklist

By placing an x in the boxes I certify that I have:

- [x] Ran `cargo test --all` locally if this modified any rs files.
- [x] Ran `cargo +stable fmt --all` locally if this modified any rs files.
- [x] Updated the content of the book if this PR would make the book outdated. (ran global regex, so hopefully)
- [x] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [x] Added unit tests for new APIs if any were added in this PR.
- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

### CODE REVIEW

File: amethyst_core/src/transform/components/local_transform.rs [hunk 1]
```diff
@@ -1,15 +1,15 @@
 //! Local transform component.
 use std::fmt;
 
-use nalgebra::{
+use crate::ecs::prelude::{Component, DenseVecStorage, FlaggedStorage};
+use crate::math::{
```
Comment (azriel91) [inline amethyst_core/src/transform/components/local_transform.rs]: these two could merge

File: amethyst_core/src/transform/components/transform.rs [hunk 2]
```diff
@@ -2,9 +2,9 @@
 
 use std::borrow::Borrow;
 
-use nalgebra::{self as na, Matrix4};
+use crate::ecs::prelude::{Component, DenseVecStorage, FlaggedStorage};
+use crate::math::{self as na, Matrix4};
```
Comment (azriel91) [inline amethyst_core/src/transform/components/transform.rs]: these can merge too!

File: amethyst_core/src/transform/systems.rs [hunk 3]
```diff
@@ -145,9 +145,9 @@ impl<'a> System<'a> for TransformSystem {
 
 #[cfg(test)]
 mod tests {
-    use nalgebra::{Matrix4, Quaternion, Unit};
+    use crate::ecs::prelude::{Builder, World};
+    use crate::math::{Matrix4, Quaternion, Unit};
```
Comment (azriel91) [inline amethyst_core/src/transform/systems.rs]: merge 3

File: docs/CHANGELOG.md [hunk 4]
```diff
@@ -45,6 +45,7 @@ it is attached to. ([#1282])
 * `FrameRateLimitConfig` has a `new` constructor, and its fields are made public. ([#1436])
 * Derive `Deserialize, Serialize` for `MaterialPrimitive` and `SpriteRenderPrimitive`, remove
 extra bounds from `AnimatablePrefab` and `AnimationSetPrefab` ([#1435])
+* Renamed amethyst_core::specs to amethyst_core::ecs and amethyst_core::nalgebra to amethyst_core::math. ([#1410])
```
Comment (azriel91) [inline docs/CHANGELOG.md]:
```md
* Renamed `amethyst_core::specs` to `amethyst_core::ecs` and `amethyst_core::nalgebra` to `amethyst_core::math`. ([#1410])
```

**Comment by jojolepro** [comment]
@azriel91 r? :)

**Comment by azriel91** [comment]
btw @jojolepro all ✅ in first post?

also, I wanna r+ <https://github.com/amethyst/amethyst/pull/1410> (re-export `nalgebra` as `math`, `specs` as `ecs`)
but since it's a bigger breaking change, I'll let it sit for a day and ping internal in case someone has an objection

**Comment by jojolepro** [comment]
I ticked the tickable boxes :P

**Comment by jojolepro** [comment]
bors r=Moxinilian,azriel91,Xaeroxe

**Pull request merged**